tornado 4.4.1 test failure

Trying to update the Fedora package to 4.4.1:
```
======================================================================
FAIL: test_parse_config_file (tornado.test.options_test.OptionsTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tornado/test/options_test.py", line 44, in test_parse_config_file
    self.assertEqual(options.my_path, config_path)
AssertionError: '/builddir/build/BUILD/tornado-4.4.1/tornado/test/options_test.cfg' != 'tornado/test/options_test.cfg'
``
Only for python2.  python3 tests run fine.  See https://kojipkgs.fedoraproject.org//work/tasks/4339/15014339/build.log

--- a/.travis.yml
+++ b/.travis.yml
@@ -30,9 +30,13 @@ install:
     - curl-config --version; pip freeze
 
 script:
-    # Get out of the source directory before running tests to avoid PYTHONPATH
-    # confusion.  This is necessary to ensure that the speedups module can
-    # be found in the installation directory.
+    # Run the tests once from the source directory to detect issues
+    # involving relative __file__ paths; see
+    # https://github.com/tornadoweb/tornado/issues/1780
+    - unset TORNADO_EXTENSION && python -m tornado.test
+    # For all other test variants, get out of the source directory before
+    # running tests to ensure that we get the installed speedups module
+    # instead of the source directory which doesn't have it.
     - cd maint
     # Copy the coveragerc down so coverage.py can find it.
     - cp ../.coveragerc .

--- a/tornado/test/options_test.py
+++ b/tornado/test/options_test.py
@@ -36,7 +36,7 @@ class OptionsTest(unittest.TestCase):
         options.define("port", default=80)
         options.define("username", default='foo')
         options.define("my_path")
-        config_path = os.path.join(os.path.dirname(__file__),
+        config_path = os.path.join(os.path.dirname(os.path.abspath(__file__)),
                                    "options_test.cfg")
         options.parse_config_file(config_path)
         self.assertEqual(options.port, 443)

